### ISSUE TITLE
[node] musl broken for 8.0.0

### ISSUE THREAD

**Issue by angerman** [input-output-hk/cardano-node#5213]
For some reason our musl based builds on `master` do build, but not run correctly.
e.g.
```bash
nix run .#hydraJobs.musl.cardano-node  -- run --config /nix/store/2s35gravj0i0d87h7zg21ihrcjcy6r5v-config-0-0.json --database-path state-node-preview/db-preview --topology /nix/store/w1ch432qw57lc460qp5sm4wn8d16yrrv-topology.yaml --host-addr 0.0.0.0 --port 3001 --socket-path state-node-preview/node.socket +RTS -N2 -I0 -A16m -qg -qb --disable-delayed-os-memory-return -RTS
```
will crash with:
```
[CLR:cardano.node:Critical:5] [2023-04-10 17:33:00.17 UTC] fromList [("error",String "could not create initial ChainValidationState: InvalidCertificate\nCallStack (from HasCallStack):\n error, called at src
/Ouroboros/Consensus/Byron/Ledger/Ledger.hs:133:19 in ouroboros-consensus-byron-0.1.0.1-A6Uz6cpocu3BICcYhLnc6M:Ouroboros.Consensus.Byron.Ledger.Ledger"),("kind",String "ConsensusStartupException")] 
 
Shutting down.. 
 
cardano-node: could not create initial ChainValidationState: InvalidCertificate 
CallStack (from HasCallStack): 
 error, called at src/Ouroboros/Consensus/Byron/Ledger/Ledger.hs:133:19 in ouroboros-consensus-byron-0.1.0.1-A6Uz6cpocu3BICcYhLnc6M:Ouroboros.Consensus.Byron.Ledger.Ledger
```

@andreabedini did some bisecting:
There's only, 4 commits between 1.35.7 and 1.35.6, so I tried them all, with 
```
nix run github:input-output-hk/cardano-node/$REF#hydraJobs.x86_64-linux.musl.cardano-node -- run --database-path $(mktemp -d) --socket-path ~/cardano-temp/node.socket --config ~/cardano-temp/config/config.json --topology ~/cardano-temp/config/topology.json
```

:x: f0b4ac897 (tag: 1.35.7-rc1, tag: 1.35.7) bad
:x: 930781a6e bad
:warning: 7529a1853 doesn't build
:warning: e389def26 doesn't build
:white_check_mark:  07b0c71d2 (tag: 1.35.6-rc2, tag: 1.35.6) good

See also:
- #5076

**Comment by CarlosLopezDeLara** [comment]
Still present on cardano-node 8.0.0 with binaries from https:// update-cardano-mainnet.iohk.io/cardano-node-releases/cardano-node-8.0.0-linux.tar.gz

cardano-node --version
cardano-node 8.0.0 - linux-x86_64 - ghc-8.10
git rev 69a117b7be3db0f4ce6d9fc5cd4c16a2a409dcb8

```
[CLR:cardano.node:Critical:5] [2023-05-10 04:08:10.61 UTC] fromList [("error",String "could not create initial ChainValidationState: InvalidCertificate\nCallStack (from HasCallStack):\n  error, called at src/byron/Ouroboros/Consensus/Byron/Ledger/Ledger.hs:130:19 in ouroboros-consensus-cardano-0.5.0.1-D3rPEw6p98936IsqPwJ3R8:Ouroboros.Consensus.Byron.Ledger.Ledger"),("kind",String "ConsensusStartupException")]

Shutting down..

cardano-node: could not create initial ChainValidationState: InvalidCertificate
CallStack (from HasCallStack):
  error, called at src/byron/Ouroboros/Consensus/Byron/Ledger/Ledger.hs:130:19 in ouroboros-consensus-cardano-0.5.0.1-D3rPEw6p98936IsqPwJ3R8:Ouroboros.Consensus.Byron.Ledger.Ledger


```

**Comment by ADADarkpool** [comment]
I am experiencing this as well.  Ubuntu 22.04, upgrading to v8.0.0. 

Using the config files provided https://book.world.dev.cardano.org/environments.html

Tried with fresh database also.

**Comment by spireblockchain** [comment]
Same issue here with Ubuntu 20.04 and cardano-node 8.0.0.  Binaries from https://update-cardano-mainnet.iohk.io/cardano-node-releases/cardano-node-8.0.0-linux.tar.gz.  Reverted to 1.35.6.

**Comment by angerman** [comment]
A temporary fix for this is in https://github.com/input-output-hk/cardano-node/pull/5223. 

A proper fix is in the works and tracked in devx-aux#39
